asmcomp/s390x/emit.mlp

```diff
@@ -436,7 +436,28 @@ let emit_instr i =
         `	brcl    4, {emit_label lbl_call_gc}\n`;  (* less than *)
         `{emit_label lbl_after_alloc}:`;
         `	la      {emit_reg i.res.(0)}, 8(%r11)\n`
-
+  | Lop(Ipoll { return_label }) ->
+      let offset = Domainstate.(idx_of_field Domain_young_limit) * 8 in
+        `	clg	%r11, {emit_int offset}(%r10)\n`;
+      let lbl_call_gc = new_label () in
+      let label_after_gc = match return_label with
+                  | None -> new_label()
+                  | Some(lbl) -> lbl in
+      let lbl_frame =
+        record_frame_label i.live (Dbg_alloc [])
+      in
+      begin match return_label with
+      | None -> `	brcl    13, {emit_label lbl_call_gc}\n`;  (* less than *)
+      | Some return_label -> `	brcl    2, {emit_label return_label}\n`;  (* higher *)
```

Comment:
By analogy with the `Ialloc` code, "less than" is 4, that is, 0b0100.  The opposite is not "higher" but "greater or equal" and is 0b1010, that is, 10.  See the handy comment for `branch_for_comparison`:
```
(* bit 0 = eq, bit 1 = lt, bit 2 = gt, bit 3 = overflow*)
```
and remember that IBM writes "bit 0" for the most significant bit.